java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -195,11 +216,37 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
     for (int i = 0; i < nestedColumnVectors.size(); i++) {
       childHandles[i] = nestedColumnVectors.get(i).getViewHandle();
     }
-    offHeap = new OffHeapState(type, (int) rows, nullCount, dataBuffer, validityBuffer, offsetBuffer,
```

Comment:
This, and every other constructor/method, that takes a `DType` rather than a `DataType` should assert/throw if it's a decimal type since that's an incomplete type.

Using something like the suggested `toDataType` method and implementing all existing `DType` methods in terms of their `DataType` versions would cover that.  We probably should also deprecate all the existing `DType` methods, pointing to the `DataType` version instead.

Reply:
I agree and we should look at each API and consider deprecating the versions that only take a DType.